node/libs/roles/src/validator/messages/consensus.rs

```diff
@@ -74,95 +75,132 @@ impl Default for Fork {
+                .collect(),
+            indexes: set
+                .clone()
+                .into_iter()
+                .enumerate()
+                .map(|(i, pk)| (pk, i))
+                .collect(),
+            total_weight,
         })
     }
 
-    /// Iterates over validators.
-    pub fn iter(&self) -> impl Iterator<Item = &validator::PublicKey> {
+    /// Iterates over weighted validators.
+    pub fn iter(&self) -> impl Iterator<Item = &WeightedValidator> {
         self.vec.iter()
     }
 
+    /// Iterates over validator keys.
+    pub fn iter_keys(&self) -> impl Iterator<Item = &validator::PublicKey> {
+        self.vec.iter().map(|v| &v.key)
+    }
+
     /// Returns the number of validators.
-    #[allow(clippy::len_without_is_empty)] // a valid `ValidatorSet` is always non-empty by construction
+    #[allow(clippy::len_without_is_empty)] // a valid `ValidatorCommittee` is always non-empty by construction
     pub fn len(&self) -> usize {
         self.vec.len()
     }
 
-    /// Returns true if the given validator is in the validator set.
+    /// Returns true if the given validator is in the validator committee.
     pub fn contains(&self, validator: &validator::PublicKey) -> bool {
-        self.map.contains_key(validator)
+        self.indexes.contains_key(validator)
     }
 
-    /// Get validator by its index in the set.
-    pub fn get(&self, index: usize) -> Option<&validator::PublicKey> {
+    /// Get validator by its index in the committee.
+    pub fn get(&self, index: usize) -> Option<&WeightedValidator> {
         self.vec.get(index)
     }
 
-    /// Get the index of a validator in the set.
+    /// Get the index of a validator in the committee.
     pub fn index(&self, validator: &validator::PublicKey) -> Option<usize> {
-        self.map.get(validator).copied()
+        self.indexes.get(validator).copied()
     }
 
-    /// Computes the validator for the given view.
+    /// Computes the leader for the given view.
     pub fn view_leader(&self, view_number: ViewNumber) -> validator::PublicKey {
         let index = view_number.0 as usize % self.len();
-        self.get(index).unwrap().clone()
+        self.get(index).unwrap().key.clone()
+    }
+
+    /// Signature weight threshold for this validator committee.
+    pub fn threshold(&self) -> u64 {
+        threshold(self.total_weight())
     }
 
-    /// Signature threshold for this validator set.
-    pub fn threshold(&self) -> usize {
-        threshold(self.len())
+    /// Maximal weight of faulty replicas allowed in this validator committee.
+    pub fn max_faulty_weight(&self) -> u64 {
+        max_faulty_weight(self.total_weight())
     }
 
-    /// Maximal number of faulty replicas allowed in this validator set.
-    pub fn faulty_replicas(&self) -> usize {
-        faulty_replicas(self.len())
+    /// Compute the sum of signers weights.
+    pub fn weight(&self, signers: Signers) -> u64 {
+        self.vec
```

Comment:
add an assert checking that length of Signers matches length of self.vec. Document that it panics if lengths do not match.